### PR TITLE
Redesign: Implemented size+position panel

### LINKED ISSUE #6468: Redesign: Update size+position panel
## Feature Description
See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3399%3A99714)

- [x] Remove "Set as background"
- [x] Update inputs
- [x] Update toggles
- [x] Replace tooltips
- [x] Update all karma tests using "Set as background" to drag to canvas edge instead
- [x] Make sure all other tests still function

## Additional Context
Screenshot 2021-02-22 at 12 38 46

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Context

Implements size&position panel as per #6468.

## User-facing changes

| Before | After |
|-|-|
| Screen Shot 2021-03-07 at 14 44 25 | Screen Shot 2021-03-07 at 14 43 40 |
| Screen Shot 2021-03-07 at 14 44 39 | Screen Shot 2021-03-07 at 14 43 46 |
| Screen Shot 2021-03-07 at 23 32 33 | Screen Shot 2021-03-07 at 23 32 12 |

### QA

This PR can be tested by following these steps:

1. Select a shape element
2. Manipulate it using the size+position panel
3. Observe that all functions still work, including flip toggles
4. Select another shape element as well at a different position and angle
5. Observe all text fields correctly displaying _"Mixed"_
6. Enter values in the mixed input fields to align the two shapes in position and/or size

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6468

### CODE REVIEW

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js [hunk 1]
```diff
@@ -25,29 +25,46 @@ import { __, _x } from '@web-stories-wp/i18n';
...
+  display: grid;
+  grid-template-areas:
+    'x . . . y . .'
+    'w . d . h . l'
+    'r . . . f . .';
+  grid-template-columns: 112px 1fr 8px 1fr 112px 1fr 32px;
+  grid-template-rows: repeat(3, 36px);
+  row-gap: 16px;
+  align-items: center;
+  justify-items: start;
```
Comment (barklund) [inline assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js]: I dedided that grid was the best way to layout all of this to also be able to reduce spacing when there's a scrollbar while also keeping both rows and columns properly aligned.

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js [hunk 2]
```diff
@@ -130,137 +140,145 @@ function SizePositionPanel({
...
+        <Area area="r">
+          <NumericInput
+            suffix={<Icons.Angle />}
+            unit={_x('°', 'Degrees, 0 - 360. ', 'web-stories')}
+            value={rotationAngle}
+            min={MIN_MAX.ROTATION.MIN}
+            max={MIN_MAX.ROTATION.MAX}
+            onChange={(evt) => {
+              const value = Number(evt.target.value);
+              pushUpdate({ rotationAngle: value }, true);
```
Comment (miina) [inline assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js:263]: Hm, seems like we're using `pushUpdate` with `true` everywhere now, perhaps the logic behind it needs changing if it it's not needed without `true` anymore 🤔  (in some other PR)

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js [hunk 3]
```diff
@@ -130,137 +140,145 @@ function SizePositionPanel({
...
-            onChange={(value) =>
-              pushUpdateForObject('flip', value, DEFAULT_FLIP, true)
-            }
-            value={flip}
           />
+        </Area>
+        <Area area="l">
+          <Tooltip
+            placement={PLACEMENT.BOTTOM}
+            title={__('Constrain proportions', 'web-stories')}
```
Comment (miina) [inline assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js]: Maybe there's something simpler/more straightforward we could use here for the tooltip? Like "Lock aspect ratio"?

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js [hunk 4]
```diff
@@ -130,137 +140,145 @@ function SizePositionPanel({
...
+                } else if (isNum(newHeight * origRatio)) {
+                  newWidth = dataPixels(newHeight * origRatio);
+                }
               }
+              pushUpdate(getUpdateObject(newWidth, newHeight), true);
+            }}
+            aria-label={__('Height', 'web-stories')}
+            isIndeterminate={MULTIPLE_VALUE === height}
+            placeholder={
+              disableHeight
```
Comment (miina) [inline assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js]: For some reason I thought we had the `no-nested-ternary` rule in `eslint` on.
I'd still prefer splitting it up for readability but not a huge difference.

**Review by miina — APPROVED** [review]
Looks great, works great! 🎉 

Thanks for the amendments to the tests, too (missing selectors etc.).

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js (hunk 4 above)
Comment (barklund) [reply assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js]: Yeah, I wasn't a huge fan of this one.

Comment (barklund) [reply assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js]: Fixed in ae6fbe42b.

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js (hunk 3 above)
Comment (barklund) [reply assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js]: Fixed in 12cf20fe4.

**Comment by miina** [comment]
@googlebot I consent.

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js [hunk 5]
```diff
@@ -130,137 +140,144 @@ function SizePositionPanel({
...
+            max={MIN_MAX.WIDTH.MAX}
+            onChange={(evt) => {
+              const newWidth = Number(evt.target.value);
+              let newHeight = height;
+              if (lockAspectRatio) {
+                if (newWidth === '') {
+                  newHeight = '';
+                } else if (isNum(newWidth / origRatio)) {
+                  newHeight = dataPixels(newWidth / origRatio);
+                }
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js:204]: nit: for readability we could move this out into its own event handler function? If we wrap it in a `useCallback` we also don't have to worry about this function being re-created every render. It's a little large for an inline function imo, but it's no big deal either way!

File: assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js [hunk 6]
```diff
@@ -130,137 +140,144 @@ function SizePositionPanel({
...
           />
+        </Area>
+        <Area area="l">
+          <Tooltip
+            placement={PLACEMENT.BOTTOM}
+            title={__('Lock aspect ratio', 'web-stories')}
+          >
+            <LockToggle
+              aria-label={__('Lock aspect ratio', 'web-stories')}
+              title={__('Lock aspect ratio', 'web-stories')}
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/design/sizePosition/sizePosition.js:246]: 🤔 Should this change to `Unlock aspect ratio` when the lock is 'locked'?

**Review by samwhale — COMMENTED** [review]
Works really nicely ✨

**Pull request merged**